Update README docs for language transforms

dolfim-ibm opened the pull request:
## Why are these changes needed?

Updates for the `pdf2parquer`, `doc_chunk` and `text_encoder` transforms.

## Related issue number (if any).

https://github.com/IBM/data-prep-kit/issues/753

dolfim-ibm committed:
update pdf2parquet README

Signed-off-by: Michele Dolfi <dol@zurich.ibm.com>

dolfim-ibm committed:
add data_files_to_use

Signed-off-by: Michele Dolfi <dol@zurich.ibm.com>

dolfim-ibm committed:
doc_chunk README

Signed-off-by: Michele Dolfi <dol@zurich.ibm.com>

dolfim-ibm committed:
text_encoder README

Signed-off-by: Michele Dolfi <dol@zurich.ibm.com>

touma-I commented on transforms/language/doc_chunk/python/README.md:
@dolfim-ibm @shahrokhDaijavad What do you guys think of adding a section like this one below to show how a user can invoke the transform once they have done a pip install (alternative to cloning the repo)::

import ast
import sys

from data_processing.runtime.pure_python import PythonTransformLauncher
from data_processing.utils import ParamsUtils
from pdf2parquet_transform_python import Pdf2ParquetPythonTransformConfiguration

local_conf = {
    "input_folder": “input”,
    "output_folder": “output”,
}
params = {
    "data_local_config": ParamsUtils.convert_to_ast(local_conf),
    "data_files_to_use": ast.literal_eval("['.pdf','.docx','.pptx','.zip']"),
}
sys.argv = ParamsUtils.dict_to_req(d=params)
launcher = PythonTransformLauncher(runtime_config=Pdf2ParquetPythonTransformConfiguration())
launcher.launch()

shahrokhDaijavad replied:
Nice job, @dolfim-ibm! Great job with the README files for all three transforms. They follow the template. 

What @touma-I is suggesting would be to add these lines of code in the section that says "Code example" and has the link to the upcoming Notebook example. These lines, together with the pip install, will be used in the Notebook, but they could also be used in a Python example that is not Notebook. I am ok either way: 1) Wait for the Notebook or 2) Add the lines now. 

@dolfim-ibm Please don't pick option 1 because it will make it easier on you! Maroun's question is how useful it is to have these lines.

dolfim-ibm replied:
@touma-I @shahrokhDaijavad  I was actually adding the code block already, but then I realized it was 1-to-1 exactly the content of the example script. Instead of having to maintain multiple versions of it (with the high-risk) of being outdated, I think that linking to the example is still ok.

Honestly, I think the best is to plan in terms of a documentation engine which can embed working code examples, and to ensure in CI that those example codes are being executed.

Commit:
Merge branch 'dev' into dol-update-readme-docs

Commit:
Added notebook for pdf2parquet

Signed-off-by: Maroun Touma <touma@marouns-mbp.watson.ibm.com>

shahrokhDaijavad committed:
Update pdf2parquet.ipynb 

Made a few changes to the comment cells that explain the execution of the immediate next cell

shahrokhDaijavad committed:
Update pdf2parquet.ipynb

Restored to a valid Notebook

touma-I committed:
Added doc chunk minimal notebook

Signed-off-by: Maroun Touma <touma@us.ibm.com>

touma-I committed:
Merge branch 'dol-update-readme-docs' of github.com:IBM/data-prep-kit into dol-update-readme-docs

touma-I committed:
minimal sample notebook for how transform can be invoked

Signed-off-by: Maroun Touma <touma@us.ibm.com>

shahrokhDaijavad committed:
restoring the make venv

Signed-off-by: SHAHROKH DAIJAVAD <shahrokh@us.ibm.com>

shahrokhDaijavad committed:
unification of notebooks

Signed-off-by: SHAHROKH DAIJAVAD <shahrokh@us.ibm.com>

touma-I committed:
added constraint for pydantic to prevent  llama-index-core from picking up 2.10

Signed-off-by: Maroun Touma <touma@us.ibm.com>

shahrokhDaijavad reviewed (approved):
LGTM.
